cmd/posix-list-dir-nix.go

```diff
@@ -168,3 +160,44 @@ func readDirN(dirPath string, count int) (entries []string, err error) {
 	}
 	return
 }
+
+// getDiskUsage returns du information for the posix path, recursively.
+func getDiskUsage(diskPath string, buf []byte) (uint64, error) {
+	d, err := os.Open(diskPath)
+	if err != nil {
+		return 0, err
+	}
+	defer d.Close()
+
+	fd := int(d.Fd())
+
+	var usage uint64
+	for {
+		nbuf, err := syscall.ReadDirent(fd, buf)
+		if err != nil {
+			return 0, err
+		}
+
+		if nbuf <= 0 {
+			break
+		}
+
+		var tmpEntries []string
+		if tmpEntries, err = parseDirents(diskPath, buf[:nbuf]); err != nil {
+			return 0, err
+		}
+
+		for _, entry := range tmpEntries {
+			// Look for only file entries.
```

Comment:
@harshavardhana , implementation here is ignoring all directories and not calculating du for contents of folders.. 
```
[images/ tbuck1/ nodedetails.yaml mybucket/ .minio.sys/ myb1/ userdata.yaml json.log bucket1/ tmp/]  in disk path /home/kris/export
➜  export du bucket1
112	bucket1
```

Reply:
Yes, our `du` is not the same as filesystem `du` - our du is object storage du try basically where there are only keys and size. Rest of the things are 0 bytes.

Reply:
Oops you are talking about tree walk here, i see the problem. @poornas